docs: cleanup configuration UI

lidel opened the pull request:
This PR 
- Moves Routing to the top: the order mirrors the order of execution (Routing, Direct, then Fallback)
- Adds more details to descriptions and labels, making them more useful to people not familiar with low level details.


| BEFORE | AFTER |
| ---- | ---- |
| ![Screenshot 2025-06-06 at 17-40-54 IPFS Service Worker Gateway production@ae34360](https://github.com/user-attachments/assets/23f29fbe-e016-45e2-8294-ecbe92ea681b) | ![Screenshot 2025-06-06 at 17-39-04 IPFS Service Worker Gateway staging@d15fef1](https://github.com/user-attachments/assets/8835e3cd-b443-4a6d-abe9-e5860ebc5576) |

cc:
- https://github.com/ipshipyard/roadmaps/issues/5
- https://github.com/ipshipyard/roadmaps/issues/9
- https://github.com/ipshipyard/roadmaps/issues/4

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [ ] I have added tests that prove my fix is effective or that my feature works

lidel committed:
docs: clarify/cleanuo configuration

this moves Routing to the top, and adds more details to descriptions and
labels

SgtPooki reviewed (approved):
lgtm!

2color commented on src/pages/config.tsx:
```diff
@@ -158,47 +158,59 @@ const ConfigPage: FunctionComponent<ConfigPageProps> = () => {
+          />
+          <TextInput
+            className='e2e-config-page-input e2e-config-page-input-dnsJsonResolvers'
+            description='A newline delimited list of space delimited key+value pairs for DNS (application/dns-json) resolvers. The key is the domain suffix, and the value is the URL of the DNS resolver. Note: Use only trusted centralized gateways, as they may infer your browsing history.'
```
Is this note meant for the Recursive Gateways input?

```suggestion
            description='A newline delimited list of space delimited key+value pairs for DNS (application/dns-json) resolvers. The key is the domain suffix, and the value is the URL of the DNS resolver.'
```

SgtPooki replied:
I think a note here that tracking may be done by the gateways you use is important. Do we want to remove it fully or can we rephrase it?

lidel replied:
My bad. This type of note applies to every remote endpoint defined in config, seems I messed up adding it everywhere. 

Fixed in 6864d0b66716163ea2c9cd4d195bb65e92b05833

2color reviewed (approved):
LGTM besides one comment

lidel committed:
docs: privacy and trust notes for external URLs

addressing https://github.com/ipfs/service-worker-gateway/pull/744#discussion_r2142568728
